aten/src/ATen/cuda/LegacyTHFunctionsCUDA.cpp

```diff
@@ -362,6 +363,22 @@ Tensor & _th_masked_scatter_bool_(Tensor & self, const Tensor & mask, const Tens
     }
     return self;
 }
+
+static void alert_nondeterministic_duplicate_indices(const Tensor& indices) {
```

Comment:
Note, it would be better to do this check while the indices are being accessed in the implementation of index_copy (like I'm doing for the CPU version), to avoid this extra traversal. However, I don't think there's a very practical way to do that, considering that the CUDA implementation is split between gpu threads. Writing a duplication check algorithm that works across threads would be possible, but challenging.

Reply:
@ngimel and I recommend that, like other operations, there not be a check for duplicate indices (at least for now) and, instead, these operations are just labeled nondeterministic and prohibited